### PR TITLE
add Krona_db module

### LINKED ISSUE #988: new module: kronatools/kronadb
I think it would be good to have a module for [Krona](https://github.com/marbl/Kronal) to create a Krona database.  Based on the [local module](https://github.com/nf-core/mag/blob/master/modules/local/krona_db.nf) in nf-core/mag

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by mjakobs**
## PR checklist

Closes #988  

This pull request adds the KronaTools/kronadb module, which creates a database used by ktImportTaxonomy.   This is based on the [local module](https://github.com/nf-core/mag/blob/master/modules/local/krona_db.nf) in the nf-core/mag pipeline.  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by grst** [comment]
- ideally, you could fix the bioconda recipe, s.t. also the latest version of the container contains all required dependencies
- is there any way to specify a version for the database the tool downloads? If it always downloads the newest version, the results of two identical pipeline runs would not be stable.

**Comment by lescai** [comment]
hi there, what's the status of this? there's an approval, and all tests are passing. I'd suggest either merge or mark it as work in progress if developers want to put new changes in?

File: modules/kronatools/kronadb/main.nf [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::krona=2.7.1" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/krona:2.7.1--pl526_5"
+    } else {
+        container "quay.io/biocontainers/krona:2.7.1--pl526_5"
+    }
+
+    input:
```
Comment (drpatelh) [inline modules/kronatools/kronadb/main.nf:20]:
```suggestion
```

File: modules/kronatools/kronadb/meta.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      description: Krona Tools is a set of scripts to create Krona charts from several Bioinformatics tools as well as from text and XML files.
+      homepage: https://github.com/marbl/Krona/wiki/KronaTools
+      documentation: https://github.com/marbl/Krona/wiki/Installing
+      tool_dev_url:
+      doi: https://doi.org/10.1186/1471-2105-12-385
+      licence:
+
+input:
+  - none: There is no input.  This module downloads a pre-built taxonomy database for use with Krona Tools.
+
```
Comment (drpatelh) [inline modules/kronatools/kronadb/meta.yml:18]:
```suggestion
```

Comment (drpatelh) [inline modules/kronatools/kronadb/meta.yml:18]:
```suggestion
## There is no input.  This module downloads a pre-built taxonomy database for use with Krona Tools.

```

**Review by drpatelh — APPROVED** [review]
Looks great!

File: modules/kronatools/kronadb/meta.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+tools:
+  - kronatools:
+      description: Krona Tools is a set of scripts to create Krona charts from several Bioinformatics tools as well as from text and XML files.
+      homepage: https://github.com/marbl/Krona/wiki/KronaTools
+      documentation: https://github.com/marbl/Krona/wiki/Installing
+      tool_dev_url:
+      doi: https://doi.org/10.1186/1471-2105-12-385
+      licence:
+
+## There is no input.  This module downloads a pre-built taxonomy database for use with Krona Tools.
```
Comment (drpatelh) [inline modules/kronatools/kronadb/meta.yml:16]:
```suggestion
input:
  - none: There is no input.  This module downloads a pre-built taxonomy database for use with Krona Tools.

```

File: modules/kronatools/kronadb/main.nf [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::krona=2.7.1" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/krona:2.7.1--pl526_5"
+    } else {
+        container "quay.io/biocontainers/krona:2.7.1--pl526_5"
+    }
+
+    output:
```
Comment (drpatelh) [inline modules/kronatools/kronadb/main.nf:20]:
```suggestion
    input:

    output:
```

**Pull request merged**